include/lp_hash_table.h

```diff
@@ -25,6 +30,12 @@
  * Search: O(1) amortized.
  */
 template <typename T> struct LpHashTable {
+  /*
+   * The map is implemented as a linear probing hash table. Keys are hashed by their
+   * pointer and are inserted into the table at the first empty spot (found linearly)
+   * in case the spot is taken. Find is implemented in a similar fashion. Delete is
+   * not implemented as we have no need for it.
+   */
```

Comment:
Please rephrase that a little. It's generally good, but I propose 2 small changes. First, explain the fact that we have 2 parallel arrays, one for the keys, one for the values. Second, disambiguate what happens in the case the spot is taken. So, maybe something like:
```cpp
/*
 * The map is implemented as a linear probing hash table. There are 2 parallel
 * arrays, one for keys and one for values. The keys are hashed by their pointer
 * and are inserted into the table at the first empty spot. If the hashed spot
 * is empty, we insert the key there and the value in its parallel spot.
 * Otherwise, we search linearly for the next empty key spot (i.e. linear probe)
 * and do the same thing. Find is implemented in a similar fashion. Delete is
 * not implemented as we have no need for it. Notes:
 *  - The load factor is a constant 70% (the number of elements in the table is
 * constant).
 *  - The values must be pointers. This constraint is imposed in order to save
 * space, as some of it is wasted to keep a load factor of 70%.
 */
```
Feel free to change that further.

Reply:
I'm going to copy-paste that so we can close the issue and merge because I'm not going to have enough time in the coming weeks. We can improve on that later when we actually need the map.